### ISSUE TITLE
Permission for "ask every time" or "only this time" on Android 11 (API 30)

### ISSUE THREAD

**Issue by uyit14** [Baseflow/flutter-geolocator#1098]
### Expected behavior
 - On Android 11, when user selected "Only this time" permission, at the next time open, the app should return LocationPermission.askAgain or LocationPermission.needToAsk, ... instead LocationPermission.whileInUse. Because i think LocationPermission.whileInUse only for "While using the app" permission.
### Reproduction steps
1. Install the app and select "Only this time" permission
2. Kill the app
3. Open again
4. The app not return LocationPermission.whileInUse

### Configuration

**Version:** 6.2.0

**Platform:** 
- [ ] :robot: Android 11

**Comment by mvanbeusekom** [comment]
Hi @uyit14,

Unfortunately this is not possible on Android. When checking permission status Android will only inform us if permissions are `GRANTED` or `DENIED`. The Android SDK doesn't provide an option to distinguish between "Only this time" or "While in use". 

When the user allows "Only this time" permissions, internally Android will grant permissions to access location services and when your app is fully destroyed Android will move the permissions back to `DENIED`. Meaning the next time the app is used permissions need to be requested again. However the process for the switching the permissions on Android will take some time and if the app is restarted before the process was run the permissions will still be granted. 

So in short we cannot distinguish between `while in use` or `only this time` and not to introduce a breaking change (and keep compatibility with other platforms) we decided not to rename the `LocationPermission.whileInUse` to `LocationPermission.granted`.
